web/src/stores/sentences.ts

```diff
@@ -3,7 +3,8 @@ const contributableLocales = require('../../../locales/contributable.json') as s
 import StateTree from './tree';
 import { Sentence } from 'common';
 
-const CACHE_SET_COUNT = 10;
+const CACHE_SET_COUNT = 25;
```

Comment:
I ended up increasing the cache_count to 25, because after you submitted the first 5 sentences, the subsequent query would try to get another 5 sentences, find 7 unread target-segment sentences, and then append them, so that you end up with repeats anyway.